src/ngx_http_lua_directive.c

```diff
@@ -73,84 +73,38 @@ enum {
 char *
 ngx_http_lua_shared_dict(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
 {
-    ngx_http_lua_main_conf_t   *lmcf = conf;
-
     ngx_str_t                  *value, name;
-    ngx_shm_zone_t             *zone;
-    ngx_shm_zone_t            **zp;
-    ngx_http_lua_shdict_ctx_t  *ctx;
     ssize_t                     size;
-
-    if (lmcf->shdict_zones == NULL) {
-        lmcf->shdict_zones = ngx_palloc(cf->pool, sizeof(ngx_array_t));
-        if (lmcf->shdict_zones == NULL) {
-            return NGX_CONF_ERROR;
-        }
-
-        if (ngx_array_init(lmcf->shdict_zones, cf->pool, 2,
-                           sizeof(ngx_shm_zone_t *))
-            != NGX_OK)
-        {
-            return NGX_CONF_ERROR;
-        }
-    }
+    ngx_int_t                   rc;
 
     value = cf->args->elts;
+    name = value[1];
 
-    ctx = NULL;
-
-    if (value[1].len == 0) {
+    if (name.len == 0) {
         ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
-                           "invalid lua shared dict name \"%V\"", &value[1]);
+                           "invalid lua shared dict name \"%V\"", &name);
         return NGX_CONF_ERROR;
     }
 
-    name = value[1];
-
     size = ngx_parse_size(&value[2]);
 
-    if (size <= 8191) {
+    if (size <= NGX_HTTP_LUA_SHDICT_MINSIZE) {
         ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
                            "invalid lua shared dict size \"%V\"", &value[2]);
         return NGX_CONF_ERROR;
     }
 
-    ctx = ngx_pcalloc(cf->pool, sizeof(ngx_http_lua_shdict_ctx_t));
-    if (ctx == NULL) {
-        return NGX_CONF_ERROR;
-    }
-
-    ctx->name = name;
-    ctx->main_conf = lmcf;
-    ctx->log = &cf->cycle->new_log;
-
-    zone = ngx_http_lua_shared_memory_add(cf, &name, (size_t) size,
-                                          &ngx_http_lua_module);
-    if (zone == NULL) {
-        return NGX_CONF_ERROR;
-    }
-
-    if (zone->data) {
-        ctx = zone->data;
-
-        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
-                           "lua_shared_dict \"%V\" is already defined as "
-                           "\"%V\"", &name, &ctx->name);
-        return NGX_CONF_ERROR;
-    }
-
-    zone->init = ngx_http_lua_shdict_init_zone;
-    zone->data = ctx;
+    rc = ngx_http_lua_shared_dict_add(cf, &name, size);
+    if (rc != NGX_OK) {
+        if (rc == NGX_DECLINED) {
+            ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                               "lua_shared_dict \"%V\" is already defined as "
+                               "\"%V\"", &name, &name);
+        }
 
-    zp = ngx_array_push(lmcf->shdict_zones);
-    if (zp == NULL) {
         return NGX_CONF_ERROR;
     }
 
-    *zp = zone;
-
-    lmcf->requires_shm = 1;
```

Comment:
This was duplicated (as in the master branch) - we now only set it from `ngx_http_lua_shared_memory_add`.